amy/trainings/tests/test_filters.py

```diff
@@ -0,0 +1,264 @@
+        )
+        self.get_involved = TrainingRequirement.objects.create(
+            name="Get Involved",
+            url_required=False,
+            event_required=False,
+            involvement_required=True,
+        )
+        self.url_and_date_required, _ = Involvement.objects.get_or_create(
+            name="GitHub Contribution",
+            defaults={
+                "display_name": "GitHub Contribution",
+                "url_required": True,
+                "date_required": True,
+            },
+        )
+        self.notes_required, _ = Involvement.objects.get_or_create(
+            name="Other without date",
+            defaults={
+                "display_name": "Other without date",
+                "notes_required": True,
+                "date_required": False,
+            },
+        )
+
+        # add training progress for Spiderman
+        # training request, event task, and welcome session
+        self.training_request = TrainingRequest.objects.create(
+            person=self.spiderman,
+            review_process="open",
+            personal="Peter",
+            family="Parker",
+            email="peter@webslinger.net",
+            state="a",
+        )
+        self.event = Event.objects.create(
+            start=datetime.today() + timedelta(days=-31),
+            end=datetime.today() + timedelta(days=-30),
+            slug="training-event-ttt",
+            host=self.org_alpha,
+            url="http://example.org/training-event-ttt",
+            country="US",
+            venue="School",
+            address="Overthere",
+            latitude=1,
+            longitude=2,
+        )
+        Task.objects.create(
+            event=self.event,
+            person=self.spiderman,
+            role=Role.objects.get(name="learner"),
+        )
+        self.progress1 = TrainingProgress.objects.create(
+            requirement=self.welcome,
+            trainee=self.spiderman,
+            state="p",
+        )
+
+        # add some Get Involved progress to another trainee
+        # (pretending that it was submitted by them)
+        self.progress2 = TrainingProgress.objects.create(
+            requirement=self.get_involved,
+            involvement_type=self.url_and_date_required,
+            trainee=self.ironman,
+            state="n",
+            date=datetime(2023, 5, 31),
+            url="https://example.com",
+        )
+
+        # add progress for a trainee who has their instructor badge
+        self.progress3 = TrainingProgress.objects.create(
+            requirement=self.welcome,
+            trainee=self.hermione,
+            state="p",
+        )
+
+        # get filterset
+        self.filterset = TraineeFilter({})
+
+        # use a queryset where trainee annotations are available (e.g. is_instructor)
+        # this still includes all Persons
+        self.qs = all_trainees_queryset()
+
+    def tearDown(self) -> None:
```

Comment:
I don't think you have to remove anything from the database. The tests are supposed to be run in nested transaction (pseudo-code):
```
TRANSACTION;
setUp()
TRANSACTION;
runTest1();
ROLLBACK;
TRANSACTION;
runTest2();
ROLLBACK;
ROLLBACK;
```